governance/program/src/state/proposal.rs

```diff
@@ -88,6 +88,12 @@ pub struct Proposal {
     /// Note: This field is not used in the current version
     pub execution_flags: InstructionExecutionFlags,
 
+    /// The supply of the Governing Token mint at the time Proposal entered voting stage
+    /// It's used as the reference total supply to calculate voting results
+    /// It's captured on the Proposal to prevent moving goal posts when the Proposal is being voted on
+    /// and to show correct vote results for historical proposals once the mint supply changes
+    pub governing_token_mint_supply: Option<u64>,
```

Comment:
Will you also need to capture each token owner's amount as well at the start of voting?  Or is that meant for a phase 2 item?  Otherwise, it seems like this makes it easier to fudge the vote, for example, if you mint a bunch of tokens after this is recorded.

Reply:
Thank you, you are completely right. If we could get the snapshot of the token owner's accounts we could also get the snapshot of the mint at the same time and this is what `voting_at_slot` is for (once we can use snapshots).
I reverted the changes and renamed the field to `governing_token_mint_vote_supply` to capture the mint supply at the time vote was decided. It's used to show correct vote results for decided votes. The common use case when it was causing issues was for votes on proposals to mint governing tokens. After executing it the calculations on the UI were instantaneously wrong.